### PR TITLE
LUCENE-8143: Remove no-op SpanBoostQuery

### PR DESCRIPTION
**Pull request by romseygeek**
Boosts are ignored on inner span queries, and top-level boosts can
be applied by using a normal BoostQuery, so SpanBoostQuery
itself is redundant and trappy.  This commit removes it entirely.

### CODE REVIEW

**Comment by romseygeek** [comment]
I added some tests to the complex query parser and xml query parser to ensure that top-level boosts are still correctly applied.  Other than that, this is pretty simple, mostly just removing now-redundant tests.

**Review by mikemccand — APPROVED** [review]
Looks great, thanks @romseygeek!

Do we need a `MIGRATE.txt` entry explaining the situation?

**Comment by romseygeek** [comment]
> Do we need a MIGRATE.txt entry explaining the situation?

Yes, and a CHANGES as well, which I always forget to do.  Thanks!

File: lucene/queryparser/src/test/org/apache/lucene/queryparser/xml/TestCoreParser.java [hunk 1]
```diff
@@ -201,6 +206,39 @@ public void testPointRangeQueryWithoutRange() throws ParserException, IOExceptio
     dumpResults("PointRangeQueryWithoutRange", q, 5);
   }
 
+  public void testSpanBoosts() throws Exception {
```
Comment (dsmiley) [inline lucene/queryparser/src/test/org/apache/lucene/queryparser/xml/TestCoreParser.java:209]: Nice testing!

**Pull request merged**